atlp-rwanda/falcons-ec-be
# #184637260 ft(): Mark product available/unavailable

**What does this PR do?**

This PR adds a feature that allows sellers to to mark product available/unavailable in their collection
**Description of Task to be completed?**
Products should be availed for buyers to purchase by default when a product is added is unavailable. A seller should be able to mark a product as available or unavailable manually in order to make it available for the buyers on not.
Reasons can be expiry date, low on stock, or they no longer deal with the product in that line, etc.


How should this be manually tested?

- clone the repository
- checkout ft-mark-product-availability-184637260 
- Run npm install to install the dependencies and dev dependencies in your terminal
- Run npm run dev to start the server in terminal and check if server is working
- In URL field enter localhost:PORT/api-docs to open swagger UI
- Login users using /api/v1/users/signin endpoint
- Verify your toke with otp at /api/v1/users/otp/verify/{token} endpoint
- Authorize /api/v1/products/{id}/availability using your verified token in the authorization key
- Mark a product available/unavailable using /api/v1/products/{id}/availability endpoint

**Any background context you want to provide?**
none
**What are the relevant pivotal tracker/Trello stories?**
tracker : #184637260 
**Screenshots (if appropriate)**
none
**Questions:**
none
